toxencryptsave/data_description.hpp

```diff
@@ -0,0 +1,43 @@
+#ifndef TOXENCRYPTSAVE_DATA_DESCRIPTION_H
+#define TOXENCRYPTSAVE_DATA_DESCRIPTION_H
+
+#include "constants.hpp"
+
+namespace toxencryptsave
+{
+
+namespace offsets
+{
+    static const size_t base = 0;
+    static const size_t magic = base;
+    static const size_t salt = magic + constants::magic_len;
+    static const size_t nonce = salt + constants::salt_len;
+    static const size_t crypted_raw = nonce + constants::nonce_len;
+};
+
+template<class PointerT>
+struct Data_Description /* encryptsave data offsets + length */
```

Comment:
Should we really be concerned about this optimizations and perform all this inplace pointer manipulations over data? I think, it would be much safer to just copy data. Making a copy shouldn't be a bottleneck.